cellprofiler/modules/tram.py

```diff
@@ -0,0 +1,673 @@
+            "Number of spline knots", 4, minval=MIN_NUM_KNOTS, doc="""
+            Number of spline knots</i>The number of knots (indpendent values) used
+            when computing smoothing splines. This should be around 1/5th the number
+            of frames for reasonably oversampled time lapse sequences, and must be 3
+            or greater. It is approximately the maximum number of wiggles expected in
+            well-tracked trajectories
+            """)
+
+        # TrAM exponent
+        self.p = cps.Float(
+            "TrAM exponent", 0.5, minval=0.01, maxval=1, doc="""
+            This number is between 0.01 and 1 (default 0.5), and specifies how
+            strongly simultaneous sudden changes in multiple features synergize in
+            the TrAM metric. A lower value signifies higher synergy (at the risk of
+            missing tracking failures that are reflected in only some of the features).
+            """)
+
+    def settings(self):
+        return [self.object_name, self.tram_measurements, self.wants_XY_Euclidian, self.num_knots, self.p]
+
+    def validate_module(self, pipeline):
+        '''Make sure that the user has selected at least one measurement for TrAM and that there are tracking data.'''
+        if len(self.get_selected_tram_measurements()) == 0:
+            raise cps.ValidationError(
+                    "Please select at least one TrAM measurement for tracking of %s" % self.object_name.get_value(),
+                    self.tram_measurements)
+
+        # check on available tracking columns for the selected object
+        obj_name = self.object_name.get_value()
+        mc = pipeline.get_measurement_columns()
+        num_tracking_cols = len([entry for entry in mc if entry[0] == obj_name and entry[1].startswith(trackobjects.F_PREFIX)])
+        if num_tracking_cols == 0:
+            msg = "No %s data available for %s. Please select an object with tracking data."\
+                  % (trackobjects.F_PREFIX, obj_name)
+            raise cps.ValidationError(msg, self.object_name)
+
+    def visible_settings(self):
+        return self.settings()
+
+    def run(self, workspace):
+        """Run the module (abstract method)
+
+        workspace    - The workspace contains
+            pipeline     - instance of cpp for this run
+            image_set    - the images in the image set being processed
+            object_set   - the objects (labeled masks) in this image set
+            measurements - the measurements for this run
+            frame        - the parent frame to whatever frame is created.
+                           None means don't draw.
+            display_data - the run() module should store anything to be
+                           displayed in this attribute, which will be used in
+                           display()
+
+        run() should not attempt to display any data, but should communicate it
+        to display() via the workspace.
+        """
+        pass
+
+
```

Comment:
Unnecessary new lines. Only one line is required between the end of one method and the beginning of the next.